libdns.go

```diff
@@ -1,394 +1,212 @@
-// Therefore, it is invalid to use this method with “CNAME”-type records.
-//
-// Implementations must honor context cancellation and be safe for concurrent
-// use.
 type RecordAppender interface {
+	// AppendRecords creates the inputted records in the given zone and returns
+	// the populated records that were created. It never changes existing records.
+	// Therefore, it is invalid to use this method with CNAME-type records.
+	//
+	// Implementations must honor context cancellation and be safe for concurrent
+	// use.
 	AppendRecords(ctx context.Context, zone string, recs []Record) ([]Record, error)
 }
 
 // [RecordSetter] can set new or update existing records in a DNS zone.
-//
-// “SetRecords” updates the zone so that the records described in the input are
-// reflected in the output. It may create or overwrite records or—depending on
-// the record type—delete records to maintain parity with the input. No other
-// records are affected. It returns the records which were set.
-//
-// For any (name, type) pair in the input, “SetRecords” ensures that the only
-// records in the output zone with that (name, type) pair are those that were
-// provided in the input.
-//
-// In RFC 7719 terms, “SetRecords” appends, modifies, or deletes records in the
-// zone so that for each RRset in the input, the records provided in the input
-// are the only members of their RRset in the output zone.
-//
-// Implementations may decide whether or not to support DNSSEC-related records
-// in calls to “SetRecords”, but should document their decision. Note that the
-// decision to support DNSSEC records in “SetRecords” is independent of the
-// decision to support them in [libdns.RecordGetter.GetRecords], so end-users
-// should not blindly call “SetRecords” on the output of
-// [libdns.RecordGetter.GetRecords].
-//
-// Implementations must honor context cancellation and be safe for concurrent
-// use.
-//
-// # Examples
-//
-// Example 1:
-//
-//	;; Original zone
-//	example.com. 3600 IN A   192.0.2.1
-//	example.com. 3600 IN A   192.0.2.2
-//	example.com. 3600 IN TXT "hello world"
-//
-//	;; Input
-//	example.com. 3600 IN A   192.0.2.3
-//
-//	;; Resultant zone
-//	example.com. 3600 IN A   192.0.2.3
-//	example.com. 3600 IN TXT "hello world"
-//
-// Example 2:
-//
-//	;; Original zone
-//	a.example.com. 3600 IN AAAA 2001:db8::1
-//	a.example.com. 3600 IN AAAA 2001:db8::2
-//	b.example.com. 3600 IN AAAA 2001:db8::3
-//	b.example.com. 3600 IN AAAA 2001:db8::4
-//
-//	;; Input
-//	a.example.com. 3600 IN AAAA 2001:db8::1
-//	a.example.com. 3600 IN AAAA 2001:db8::2
-//	a.example.com. 3600 IN AAAA 2001:db8::5
-//
-//	;; Resultant zone
-//	a.example.com. 3600 IN AAAA 2001:db8::1
-//	a.example.com. 3600 IN AAAA 2001:db8::2
-//	a.example.com. 3600 IN AAAA 2001:db8::5
-//	b.example.com. 3600 IN AAAA 2001:db8::3
-//	b.example.com. 3600 IN AAAA 2001:db8::4
 type RecordSetter interface {
+	// SetRecords updates the zone so that the records described in the input are
```

Comment:
I suppose it's out of the question to rename this method to SetRecordSets? It should make it clear that the behaviour is quite a bit different/more special than the Append and Delete methods. I'm afraid users/implementers of this method may make the wrong assumption based on the name...

Reply:
Hm, yeah I dunno, I like the current naming convention because it's consistent.

I also like the "play" on words: Set refers to both the noun and verb form. (Makes a lot of sense when you think about it, set applies the definition of a set.)

Reply:
> I suppose it's out of the question to rename this method to SetRecordSets?

I think that if you know enough about DNS to know what an RRset is, you would know to be suspicious of methods named `SetRecords` and fully read their documentation before using them. `SetRecordSets` would definitely be a more correct name, but I still don't think that that name makes it obvious what the method does without first reading its documentation.